raiden/api/python.py

```diff
@@ -651,26 +651,50 @@ def get_channel_events(
                 from_block=from_block,
                 to_block=to_block,
             ))
+        returned_events.sort(key=lambda evt: evt.get('block_number'), reverse=True)
+        return returned_events
 
+    def get_channel_events_raiden(
+            self,
+            token_address: typing.Address,
+            partner_address: typing.Address = None,
+            from_block: typing.BlockSpecification = 0,
+            to_block: typing.BlockSpecification = 'latest',
+    ):
+        returned_events = []
         raiden_events = self.raiden.wal.storage.get_events_by_block(
             from_block=from_block,
             to_block=to_block,
         )
-
         # Here choose which raiden internal events we want to expose to the end user
         for block_number, event in raiden_events:
-            if isinstance(event, EVENTS_EXTERNALLY_VISIBLE):
-                new_event = {
-                    'block_number': block_number,
-                    'event': type(event).__name__,
-                }
-                new_event.update(event.__dict__)
-                returned_events.append(new_event)
+            new_event = {
+                'block_number': block_number,
+                'event': type(event).__name__,
+            }
+            if hasattr(event, 'transfer'):
+                if event.transfer.token == token_address:
+                    event.transfer = event.transfer.__repr__()
```

Comment:
Please, use `repr(obj)` instead of `obj.__repr__()`.

Also, why is this being done?

Reply:
Because I could not return the object after I've used encode_byte_values: I would always get... Object of type ‘<something>’ is not JSON serializable. So I decided to use the __repr__. Anyway, this is gonna be used only for debugging.

Reply:
IMO this is not a good solution. I would prefer to have the events encoded using marshmallow so that we can have proper json output.

to do this you should encode the output at the REST API level, [example](https://github.com/raiden-network/raiden/blob/2291e863273763983f2144a9078758e9cb81babb/raiden/api/rest.py#L449), and create the missing schemas [example](https://github.com/raiden-network/raiden/blob/2291e863273763983f2144a9078758e9cb81babb/raiden/api/v1/encoding.py#L244)